zerver/lib/bugdown/__init__.py

```diff
@@ -1825,9 +1825,15 @@ def __init__(self, realm_id: int, content: str) -> None:
 
         user_group_names = possible_user_group_mentions(content)
         self.user_group_name_info = get_user_group_name_info(realm_id, user_group_names)
+        self.user_group_members = defaultdict(list)  # type: Dict[int, List[int]]
         group_ids = [group.id for group in self.user_group_name_info.values()]
+
+        if not group_ids:
+            # Early-return to avoid the cost of hitting the ORM,
+            # which shows up in profiles.
+            return
+
         membership = UserGroupMembership.objects.filter(user_group_id__in=group_ids)
-        self.user_group_members = defaultdict(list)  # type: Dict[int, List[int]]
         for info in membership.values('user_group_id', 'user_profile_id'):
             group_id = info['user_group_id']
             user_profile_id = info['user_profile_id']
```

Comment:
I'm worried about this code resulting in folks adding code below here, not realizing it won't run without `group_ids`.  Maybe move this `group_ids` block into a function rather than doing early return?

Reply:
I am extracting a function here.